### PR TITLE
Draft PR : Tutorial for Lagrangian Neural Network

### PR DESCRIPTION
**Pull request by a-b-h-a-y-s-h-i-n-d-e**
## Description

This PR is currently in draft state
following are the topics/sections for LNN tutorial 
* Introduction
* compare NN with LNN
* about spring pendulum experiment
* about Euler lagrangian equations
* explain synthetic dataset code
* training the model
* visualizing/plotting the results

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ x ] Documentations (modification for documents)

## Checklist

- [ x  ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ x ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ x ] Run `mypy -p deepchem` and check no errors
  - [ x ] Run `flake8 <modified file> --count` and check no errors
  - [ x ] Run `python -m doctest <modified file>` and check no errors
- [ x ] I have performed a self-review of my own code
- [ x ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ x ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: examples/tutorials/Introduction_to_Lagrangian_Neural_Network.ipynb [hunk 1]
```diff
@@ -0,0 +1,467 @@
+{
```
Comment (sudo-rsingh) [inline examples/tutorials/Introduction_to_Lagrangian_Neural_Network.ipynb:1]: <p>Rather than adding links, pls use references and just give the reference index number like <code>[x]</code> where x being the index of reference. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4548/discussion/#comment-2404628263'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (sudo-rsingh) [inline examples/tutorials/Introduction_to_Lagrangian_Neural_Network.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    from scipy.integrate import odeint
</pre><p>I think there might be something similar to odeint in deepchem, can you try to use it.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4548/discussion/#comment-2404628264'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (sudo-rsingh) [inline examples/tutorials/Introduction_to_Lagrangian_Neural_Network.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #2.      '''
</pre><p>Pls write deepchem style doc string, sometimes, we also skip writing docstrings in tutorials for the sake of better readability in that case i will suggest to use more descriptive names for functions and variables.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4548/discussion/#comment-2404628266'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (sudo-rsingh) [inline examples/tutorials/Introduction_to_Lagrangian_Neural_Network.ipynb:1]: <p>pls keep the setup section at the top and also pip install scipy and anything else like matplotlib you might be using</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4548/discussion/#comment-2404628267'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (sudo-rsingh) [inline examples/tutorials/Introduction_to_Lagrangian_Neural_Network.ipynb:1]: <p><code>Consider using tensor.detach() first</code></p><p>there are some warnings in this section, i think this warning is easy to fix can you fix it also that will be a great help thanks. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4548/discussion/#comment-2404628268'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (sudo-rsingh) [inline examples/tutorials/Introduction_to_Lagrangian_Neural_Network.ipynb:1]: <p>dont just write code everywhere, add some theory also, that will be better for learning</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4548/discussion/#comment-2404628269'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (sudo-rsingh) [inline examples/tutorials/Introduction_to_Lagrangian_Neural_Network.ipynb:1]: <p><code>/tmp/ipython-input-2884541391.py:1: MatplotlibDeprecationWarning: The 'labels' parameter of boxplot() has been renamed 'tick_labels' since Matplotlib 3.9; support for the old name will be dropped in 3.11. plt.boxplot([test_dataset.y.flatten(), predictions.flatten()],</code> </p><p>there is some warning messages can you fix these</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4548/discussion/#comment-2404628271'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (sudo-rsingh) [inline examples/tutorials/Introduction_to_Lagrangian_Neural_Network.ipynb:1]: <p>pls fix the url, and if your mentor or Bharath helped in this tutorial then keep his name also as author,</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4548/discussion/#comment-2404628273'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

**Comment by a-b-h-a-y-s-h-i-n-d-e** [comment]
@sudo-rsingh thanks for reviewing it, since this is draft PR I am still arranging sections/sub-topics, the goal is to create similar to this [tutorial](https://github.com/deepchem/deepchem/blob/master/examples/tutorials/Introduction_to_Hamiltonian_Neural_Network.ipynb)
